applications/tari_console_wallet/src/init/mod.rs

```diff
@@ -340,5 +343,37 @@ pub async fn start_wallet(wallet: &mut WalletSqlite, base_node: &Peer) -> Result
         error!(target: LOG_TARGET, "Problem restarting transaction protocols: {}", e);
     }
 
+    debug!(target: LOG_TARGET, "Starting TXO validations.");
```

Comment:
These 
- `validate_txos(TxoValidationType::Unspent`
- `validate_txos(TxoValidationType::Spent`
- `validate_txos(TxoValidationType::Invalid`

should be verified to be completed successfully and go before `Restart transaction protocols` are started, similar to the base node wallet logic
- `wallet.transaction_service.restart_transaction_protocols().await`
- `wallet.transaction_service.restart_broadcast_protocols().await`

We want to ensure transaction protocols only use a fully validated TXO set, thus they should be delayed until after the prior validation. See `async fn monitor_validation_protocol(`

Reply:
Thanks for the feedback! Will do.

Reply:
Also, please implement the higher level initial sync logic in `async fn start_transaction_protocols_and_txo_validation(`, i.e. "Only start the transaction broadcast and TXO validation protocols once the ~~local~~ _connected base_ node is synced".

Reply:
I agree with Hansie on this, I think the validations and restarting of the *broadcast* protocols should be done in an asynchronous task. However, I think that the transaction negotiation protocols can be restarted without waiting for the validation to be completed. These protocols are between wallets and not between the base node and this wallet. We want wallet-to-wallet negotiation to occur even if the base node is unreachable. If there is a problem with the outputs in a given transaction being negotiated between wallets it will be picked up when they finish their negotiation and try broadcast.